filelists/CIFAR-FS/cifar_fs_preprocessing.py

```diff
@@ -0,0 +1,490 @@
+                Tbase += [(img_id, Kbase_idx) for img_id in imd_ids]
+
+        assert (len(Tbase) == nTestBase)
+
+        return Tbase
+
+    def sample_train_and_test_examples_for_novel_categories(
+            self, Knovel, nTestNovel, nExemplars, nKbase):
+        """Samples train and test examples of the novel categories.
+        Args:
+    	    Knovel: a list with the ids of the novel categories.
+            nTestNovel: the total number of test images that will be sampled
+                from all the novel categories.
+            nExemplars: the number of training examples per novel category that
+                will be sampled.
+            nKbase: the number of base categories. It is used as offset of the
+                category index of each sampled image.
+        Returns:
+            Tnovel: a list of length `nTestNovel` with 2-element tuples. The
+                1st element of each tuple is the image id that was sampled and
+                the 2nd element is its category label (which is in the range
+                [nKbase, nKbase + len(Knovel) - 1]).
+            Exemplars: a list of length len(Knovel) * nExemplars of 2-element
+                tuples. The 1st element of each tuple is the image id that was
+                sampled and the 2nd element is its category label (which is in
+                the ragne [nKbase, nKbase + len(Knovel) - 1]).
+        """
+
+        if len(Knovel) == 0:
+            return [], []
+
+        nKnovel = len(Knovel)
+        Tnovel = []
+        Exemplars = []
+        assert ((nTestNovel % nKnovel) == 0)
+        nEvalExamplesPerClass = int(nTestNovel / nKnovel)
+
+        for Knovel_idx in range(len(Knovel)):
+            imd_ids = self.sampleImageIdsFrom(
+                Knovel[Knovel_idx],
+                sample_size=(nEvalExamplesPerClass + nExemplars))
+
+            imds_tnovel = imd_ids[:nEvalExamplesPerClass]
+            imds_ememplars = imd_ids[nEvalExamplesPerClass:]
+
+            Tnovel += [(img_id, nKbase + Knovel_idx) for img_id in imds_tnovel]
+            Exemplars += [(img_id, nKbase + Knovel_idx) for img_id in imds_ememplars]
+        assert (len(Tnovel) == nTestNovel)
+        assert (len(Exemplars) == len(Knovel) * nExemplars)
+        random.shuffle(Exemplars)
+
+        return Tnovel, Exemplars
+
+    def sample_episode(self):
+        """Samples a training episode."""
+        nKnovel = self.nKnovel
+        nKbase = self.nKbase
+        nTestNovel = self.nTestNovel
+        nTestBase = self.nTestBase
+        nExemplars = self.nExemplars
+
+        Kbase, Knovel = self.sample_base_and_novel_categories(nKbase, nKnovel)
+        Tbase = self.sample_test_examples_for_base_categories(Kbase, nTestBase)
+        Tnovel, Exemplars = self.sample_train_and_test_examples_for_novel_categories(
+            Knovel, nTestNovel, nExemplars, nKbase)
+
+        # concatenate the base and novel category examples.
+        Test = Tbase + Tnovel
+        random.shuffle(Test)
```

Comment:
Set `random.seed(a=1)` for reproducability

Reply:
I think the random seed is set in train.py or test.py. That should be enough.

Reply:
![image](https://user-images.githubusercontent.com/3034832/80619400-f5889680-8a44-11ea-820a-09ce693c8786.png)
I don't think so